src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/quantized_matmuls_with_shared_weights.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <regex>
+
+#include "common_test_utils/node_builders/constant.hpp"
+#include "common_test_utils/node_builders/fake_quantize.hpp"
+#include "common_test_utils/node_builders/reshape.hpp"
+#include "openvino/openvino.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+
+enum class FQInterval { U8, I8 };
+inline std::ostream& operator<<(std::ostream& os, FQInterval interval) {
+    switch (interval) {
+    case FQInterval::U8:
+        os << "U8";
+        break;
+    case FQInterval::I8:
+        os << "I8";
+        break;
+    default:
+        OPENVINO_THROW("Unknown FQInterval");
+    }
+    return os;
+}
+
+typedef std::tuple<InputShape, InputShape, FQInterval, FQInterval> QuantizedMatMulsWithSharedWeightsParans;
+
+class QuantizedMatMulsWithSharedWeightsTest
```

Comment:
Worth to add comment why we need this test

Reply:
The comment is added, please take a look